### PR TITLE
Basic autogen support for --stripe-packages

### PR DESCRIPTION
**Pull request by aisamanra**
### Motivation
We currently do not run the `packager` when in `--autogen` mode at all. These changes allow us to run the packager but produce the same output as before by stripping off the package name when referring to constants, but also lays the groundwork for a separate "packaged autoloader" mode that generates autoloads suitable for packages.

### Test plan

See included automated tests. Added a new CLI test to verify the output with `--stripe-packages`; this can be diff'ed against the output from the other autoloader CLI test to show that the only differences are in the file paths.

### CODE REVIEW

File: main/realmain.cc [hunk 1]
```diff
@@ -215,7 +215,7 @@ void runAutogen(const core::GlobalState &gs, options::Options &opts, const autog
             for (auto result = fileq->try_pop(idx); !result.done(); result = fileq->try_pop(idx)) {
                 ++n;
                 auto &tree = indexed[idx];
-                if (tree.file.data(gs).isRBI()) {
+                if (tree.file.data(gs).isRBI() || tree.file.data(gs).isPackage()) {
```
Comment (aisamanra) [inline main/realmain.cc:218]: This just means we don't generate autoloader information for `__package.rbi` files.

File: main/realmain.cc [hunk 2]
```diff
@@ -531,7 +531,9 @@ int realmain(int argc, char *argv[]) {
             gs->suppressErrorClass(core::errors::Namer::ModuleKindRedefinition.code);
             gs->suppressErrorClass(core::errors::Resolver::StubConstant.code);
 
+            indexed = pipeline::package(*gs, move(indexed), opts, *workers);
```
Comment (aisamanra) [inline main/realmain.cc:534]: The `pipeline::package` function already handles "do not run if `--stripe-packages` is not supplied," so we don't need to do any conditional logic here.

File: packager/packager.cc [hunk 3]
```diff
@@ -545,7 +545,7 @@ ast::ParsedFile rewritePackage(core::Context ctx, ast::ParsedFile file, const Pa
     }
 
     // Sanity check: __package.rb files _must_ be typed: strict
-    if (file.file.data(ctx).strictLevel < core::StrictLevel::Strict) {
+    if (file.file.data(ctx).originalSigil < core::StrictLevel::Strict) {
```
Comment (aisamanra) [inline packager/packager.cc:548]: We set the `strictLevel` lower artificially for autogen so we don't get errors even from invalid code. That means that in autogen every `__package.rb` file would raise this error. We can sidestep that by checking the `originalSigil`, which is retained even if `strictLevel` is set lower.

**Review by jvilk-stripe — CHANGES_REQUESTED** [review]
My only hard request is to change the `isPackage()` helper, which could cause issues down the line / is more expensive than it needs to be.

Everything else looks good to me. Thanks for adding tests!

File: main/autogen/autoloader.cc [hunk 4]
```diff
@@ -10,8 +10,13 @@
 using namespace std;
 namespace sorbet::autogen {
 
+// Like the rest of autogen, the `autoloader` pass works by walking an existing tree of data and converting it to a
+// different representation before using it. In this case, it takes the `autogen::ParsedFile` representation and
+// converts it to a `DefTree`, and then emits the autoloader files based on passed-in string fragments
+
 bool AutoloaderConfig::include(const NamedDefinition &nd) const {
-    return !nd.nameParts.empty() && topLevelNamespaceRefs.find(nd.nameParts[0]) != topLevelNamespaceRefs.end();
+    return !nd.qname.nameParts.empty() &&
```
Comment (jvilk-stripe) [inline main/autogen/autoloader.cc:18]:
```suggestion
    return !nd.qname.empty() &&
```

**Review by jvilk-stripe — APPROVED** [review]
LGTM

**Comment by aisamanra** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build result here:

→ https://go/builds/bui_IMlBHuSzWpVO0F

**Comment by aisamanra** [comment]
Pay-server prework is landing so I'm merging this.

**Pull request merged**